interop/ksp/src/main/kotlin/com/squareup/kotlinpoet/ksp/utils.kt

```diff
@@ -0,0 +1,30 @@
+package com.squareup.kotlinpoet.ksp
+
+import com.squareup.kotlinpoet.ClassName
+import com.squareup.kotlinpoet.LambdaTypeName
+import com.squareup.kotlinpoet.ParameterizedTypeName
+import com.squareup.kotlinpoet.TypeName
+
+internal fun TypeName.rawType(): ClassName {
+  return findRawType() ?: throw IllegalArgumentException("Cannot get raw type from $this")
+}
+
+internal fun TypeName.findRawType(): ClassName? {
+  return when (this) {
+    is ClassName -> this
+    is ParameterizedTypeName -> rawType
+    is LambdaTypeName -> {
+      var count = parameters.size
+      if (receiver != null) {
+        count++
+      }
+      val functionSimpleName = if (count >= 23) {
+        "FunctionN"
+      } else {
+        "Function$count"
+      }
+      ClassName("kotlin.jvm.functions", functionSimpleName)
```

Comment:
Is there a multiplatform-friendly alternative?

Reply:
I'm not sure to be honest. The [spec](https://github.com/JetBrains/kotlin/blob/master/spec-docs/function-types.md) focuses mostly on JVM and agnostic impls, but doesn't explicitly call out how it works on other platforms. Seeing as we do this here and in metadata, mind if we save this for a later day?

Reply:
For sure, especially since I believe KSP only targets JVM with 1.0? Or it will support MPP?

Reply:
KSP will support MPP yeah (insofar as it will be able to run code gen for MPP projects)